### PR TITLE
Doc: document separate compilation (close #668)

### LINKED ISSUE #668: Document separate compilation.
The documentation present in [this page](https://github.com/ocsigen/js_of_ocaml/blob/master/doc/manual/src/separate-compilation.wiki) leave a bit to desire.

### PR DESCRIPTION
**Pull request by hhugo**
This reverts commit 0ff280baecbf39c6567225eecd8dc76e0ab4e0b4.

fix #668

### CODE REVIEW

**Comment by hhugo** [comment]
@Drup, would you have time to do some review ?

File: manual/separate-compilation.wiki [hunk 1]
```diff
@@ -1 +1,85 @@
...
+The general idea is generate one JavaScript file containing the
+JavaScript runtime and JavaScript files for every compilation unit (or
+library) needed to run the program. One can then link (or
+"concatenate") all individual JavaScript files together, respecting
+the order induced by dependencies, into a single JavaScript file.
+
+=== 1. Build the runtime
+
+{{{
+js_of_ocaml --runtime-only ${ALL_RUNTIME_FILES} -o my-runtime.js
```
Comment (dbuenzli) [inline manual/separate-compilation.wiki]: It's unclear here what is supposed to be in `${ALL_RUNTIME_FILES}`

File: manual/separate-compilation.wiki [hunk 2]
```diff
@@ -1 +1,85 @@
...
+}}}
+
+=== 2. Build compilation units
+
+In addtion to bytecode executables, the js_of_ocaml compiler can
```
Comment (dbuenzli) [inline manual/separate-compilation.wiki]: addtion -> addition

File: manual/separate-compilation.wiki [hunk 3]
```diff
@@ -1 +1,85 @@
...
+=== Link
+
+The final step is to link all individual JavaScript files into a
+single one.  The js_of_ocaml-compiler provides a separate tool called
+{{{jsoo_link}}} to do that.  {{{jsoo_link}}} takes a list of
+JavaScript filename as input and concatenates them all (merging
+source-maps together if needed).
+
+{{{
+jsoo_link my-runtime.js stdlib.js libname.js -o myexe.js
```
Comment (dbuenzli) [inline manual/separate-compilation.wiki]: Isn't that maybe `+stdlib.js` or you have to compile it yourself using point 2. ?

**Comment by dbuenzli** [comment]
Also maybe adding a bit more about the debugging/source maps options could be nice.

**Comment by hhugo** [comment]
Thanks a lot for the review. I've tried to addressed all your comments.

**Comment by hhugo** [comment]
> Is there any plan to maybe migrate to .mld files

I've been looking into this and I'm still trying to figure out the integration with ocsigen.org

**Comment by Drup** [comment]
@hhugo Happy to see you starting to work on documentation. The addition looks great!

**Pull request merged**